pyanaconda/kickstart_dispatcher/element.py

```diff
@@ -0,0 +1,239 @@
+        content = ""
+        if self._type == self.KickstartElementType.COMMAND:
+            content = lines[0]
+        elif self._type in (self.KickstartElementType.ADDON,
+                            self.KickstartElementType.SECTION):
+            body = "".join(self._lines)
+            if body:
+                content = "{}\n{}%end\n".format(" ".join(self._args), body)
+            else:
+                content = "{}\n%end\n".format(" ".join(self._args))
+        return content
+
+
+class KickstartElements(object):
+    """Container for storing and filtering KickstartElement objects
+
+    Preserves order of added elements.
+    """
+
+    def __init__(self):
+        self._elements = []
+
+    def append(self, element):
+        """Appends KickstartElement to the container.
+
+        :param element: element object to be appended to the container
+        :type name: KickstartElement
+        """
+        self._elements.append(element)
+
+    @property
+    def all_elements(self):
+        """List of all elements in the container.
+
+        :return: list of all elements in the container
+        :rtype: list(KickstartElement)
+        """
+
+        return list(self._elements)
+
+    def get_elements(self, commands=None, sections=None, addons=None):
+        """Returns selected elements.
+
+        :param commands: names of commands to be returned
+        :type commands: list
+        :param sections: names of sections to be returned
+        :type sections: list
+        :param addons: names of addons to be returned
+        :type addons: list
+
+        :return: list of filtered elements
+        :rtype: list(KickstartElement)
+        """
+
+        filtered_elements = []
+        for element in self._elements:
+            if element.is_command():
+                if commands and element.name in commands:
+                    filtered_elements.append(element)
+            elif element.is_addon():
+                if addons and element.name in addons:
+                    filtered_elements.append(element)
+            elif element.is_section():
+                if sections and element.name in sections:
+                    filtered_elements.append(element)
+        return filtered_elements
+
+    @staticmethod
+    def get_kickstart_from_elements(elements=None):
+        """Returns kickstart generated from elements.
+
+        :param elements: list of kickstart elements for generated kickstart
+        :type elements: list(KickstartElement)
+
+        :return: kickstart
+        :rtype: str
+        """
+        return "".join(element.content for element in elements)
+
+    def __str__(self):
+        return str(self._elements)
+
+
+class TrackedKickstartElements(KickstartElements):
+    """Container for kickstart elements with tracking."""
+    def __init__(self):
+        super().__init__()
+        self._processed_elements = set()
+
+    def get_and_process_elements(self, commands=None, sections=None, addons=None):
```

Comment:
This used to be get_elements(self, commands=None, sections=None, addons=None, process=False) in the first PR, adding process option to the parent method signature, but pylint doesn't like changing the signature in overriden method so I used this.
I am not satisfied with method name name, get_and_track_elements may be better, plus maybe changing unprocessed_elements() to untracked_elements()